### PR TITLE
IMG-2 Images comparison

### PR DESCRIPTION
**Pull request by PoletaevaSasha**

### CODE REVIEW

File: src/main/java/com/chekanova/imagetool/controller/ImageController.java [hunk 1]
```diff
@@ -42,4 +43,17 @@ public ResponseEntity<byte[]> process(@RequestParam MultipartFile file,
...
         return ResponseEntity.ok().contentType(MediaType.IMAGE_JPEG).body(byteArrayOutputStream.toByteArray());
     }
-}
+
+    @PostMapping(value = "/compare",
+            produces = MediaType.IMAGE_JPEG_VALUE)
+    public ResponseEntity<byte[]> compare(@RequestParam MultipartFile file1,
+                                          @RequestParam MultipartFile file2,
+                                          RedirectAttributes attributes) throws IOException {
+        BufferedImage image1 = ImageIO.read(new ByteArrayInputStream(file1.getBytes()));
```
Comment (artemKotovSigmaUkraine) [inline src/main/java/com/chekanova/imagetool/controller/ImageController.java]: you should keep lines 52-56 in service

File: src/main/java/com/chekanova/imagetool/service/comparison/impl/ImageComparisonServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,193 @@
+package com.chekanova.imagetool.service.comparison.impl;
+
+import com.chekanova.imagetool.service.comparison.ImageComparisonService;
+import lombok.Setter;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.awt.*;
```
Comment (artemKotovSigmaUkraine) [inline src/main/java/com/chekanova/imagetool/service/comparison/impl/ImageComparisonServiceImpl.java]: .* imports all classes from the package - import only exact classes from the package

**Comment by artemKotovSigmaUkraine** [comment]
@PoletaevaSasha  Review completed, looks good, just several comments

File: src/main/java/com/chekanova/imagetool/service/comparison/ImageComparisonService.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
+package com.chekanova.imagetool.service.comparison;
+
+import java.awt.image.BufferedImage;
+
+/**
+ * Basic interface for comparing images
+ * @author oleksandra.chekanova
+ */
+public interface ImageComparisonService {
```
Comment (artemKotovSigmaUkraine) [inline src/main/java/com/chekanova/imagetool/service/comparison/ImageComparisonService.java:9]: Interface is not required, you can create only ImageComparisonService

File: src/main/java/com/chekanova/imagetool/service/comparison/impl/ImageComparisonServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+            stack.push(new AbstractMap.SimpleImmutableEntry<>(x, y - 1));
+            stack.push(new AbstractMap.SimpleImmutableEntry<>(x + 1, y));
+            stack.push(new AbstractMap.SimpleImmutableEntry<>(x, y + 1));
+        }
+        return box;
+    }
+
+    //Left recursive function for history
+    //(In real project should remove old code)
+/*    private void buildBox(int x, int y, int w, int h, boolean[][] pix, Box box) {
```
Comment (artemKotovSigmaUkraine) [inline src/main/java/com/chekanova/imagetool/service/comparison/impl/ImageComparisonServiceImpl.java]: remove commented code, if it is not used

File: src/main/java/com/chekanova/imagetool/controller/ImageController.java (hunk 1 above)
Comment (PoletaevaSasha) [reply src/main/java/com/chekanova/imagetool/controller/ImageController.java]: Done

File: src/main/java/com/chekanova/imagetool/service/comparison/impl/ImageComparisonServiceImpl.java (hunk 2 above)
Comment (PoletaevaSasha) [reply src/main/java/com/chekanova/imagetool/service/comparison/impl/ImageComparisonServiceImpl.java]: Done

File: src/main/java/com/chekanova/imagetool/service/comparison/ImageComparisonService.java (hunk 3 above)
Comment (PoletaevaSasha) [reply src/main/java/com/chekanova/imagetool/service/comparison/ImageComparisonService.java:9]: Done as we discussed.

File: src/main/java/com/chekanova/imagetool/service/comparison/impl/ImageComparisonServiceImpl.java (hunk 4 above)
Comment (PoletaevaSasha) [reply src/main/java/com/chekanova/imagetool/service/comparison/impl/ImageComparisonServiceImpl.java]: Yes, all comments should be deleted in projects. I wrote this to you on Skype.  But this project is for education and I left it only for my own purposes and for future investigation. I changed the reason in comment.

**Pull request merged**